### ISSUE TITLE
LeoCAD for Android (Qt5 port)

### ISSUE THREAD

**Issue by Symbian9** [leozide/leocad#65]
Hope, there is a way for porting LeoCAD on Android platform.

As example, look how OpenOrienteering Mapper ported on Android
* https://github.com/OpenOrienteering/mapper/wiki/Android
* https://github.com/OpenOrienteering/superbuild

**Comment by leozide** [comment]
It would be cool to have an Android port but I don't own any Android devices. LeoCAD runs on iOS so it may be just a matter of compiling for Android and it will "work", have you tried it?

**Comment by Symbian9** [comment]
> It would be cool to have an Android port but I don't own any Android devices.

In this case F-Droid community could help. I has Android 6.0 at least and could test it, but I can't make port or write code for it.

Think, this is long term target. Will digg more in F-Droid ecosystem.

For now I will try make some UI mocaps for it.

**Comment by Symbian9** [comment]
Some docs and examples from Qt
* https://www.qt.io/mobile-app-development/
* http://doc.qt.io/qt-5/examples-android.html
* https://wiki.qt.io/How_to_Create_and_Run_Qt_Application_for_Android

**Comment by Symbian9** [comment]
Thanks to @trevorsandy, ARM support for  LeoCAD coming soon?!

> LPub3D LeoCAD Library ARM Build Test
> - https://github.com/trevorsandy/lc_lib-obs

**Comment by Symbian9** [comment]
Topic "LDraw Mobile Client"

- https://forums.ldraw.org/thread-22803.html

**Comment by Symbian9** [comment]
New topic "LDraw Mobile Client"

- https://forums.ldraw.org/thread-23663.html

**Comment by nathaneltitane** [comment]
@leozide @Symbian9 I run Android devices exclusvely and could help compile iterations of the project.
I purchased a Galaxy S10+ recently becasue of Linux on Dex and have managed to run it on LoD and compile it.

Where should i start?

**Comment by leozide** [comment]
I imagine it compiles without a lot of work, I had it compiling and running on my iPad a while ago. The problem is it looks just like the desktop version with menus and small buttons. The UI needs to be completely redesigned for small touch devices.

**Comment by nathaneltitane** [comment]
@leozide of course, if you're talking about using it directly on he device itself.
I've had it compile on arm64 recently (we had spoken about it) and a more realistic approach would be to have it over MHL or some kind of projection dock (DeX and the likes) and use it with normal peripherals.

This doesn't exclude the fact that if you're running on a large enough device (screen wise, such as an ipad or android tabled) the use of bluetooth peripherals and pens is now a common reality and should provide enough for the user to edit without any trouble.

**Comment by nathaneltitane** [comment]
> Thanks to @trevorsandy, ARM support for LeoCAD coming soon?!
> 
> > LPub3D LeoCAD Library ARM Build Test
> > 
> > * https://github.com/trevorsandy/lc_lib-obs

already built it myself a while ago, it already builds by default @Symbian9

**Comment by nathaneltitane** [comment]
adding Buf3D as a reference android application: it reads loads and parses LDraw files amazingly well and does so natively on android using the ldraw parts library. 

it may just be possible after all :)

**Comment by nathaneltitane** [comment]
> I imagine it compiles without a lot of work, I had it compiling and running on my iPad a while ago. The problem is it looks just like the desktop version with menus and small buttons. The UI needs to be completely redesigned for small touch devices.

any chance a sample build target project could be put together for me to see what gives? I could give feedback on the experience

**Comment by Symbian9** [comment]
> The UI needs to be completely redesigned for small touch devices.

There could be two UI modes, for Desktop and Mobile.

See, OpenOrienteering Mapper
- https://github.com/OpenOrienteering/mapper
  - https://www.openorienteering.org/mapper-manual/pages/touch-mode.html#using-the-mapper-app-in-touch-mode
  > The “Touch Mode” is an alternative user interface which is always active on Android devices. It can be activated on PCs, too.
